inventory.py

```diff
@@ -71,23 +72,30 @@ def slim_down_course_data(course_data: Sequence[Dict]) -> Sequence[Dict]:
 
 
 def gather_course_info_for_account(account_id: int, term_id: int) -> Sequence[int]:
-    url_ending = f'accounts/{account_id}/courses'
+    url_ending_with_scope = f'{API_SCOPE_PREFIX}/accounts/{account_id}/courses'
     params = {
         'with_enrollments': True,
         'enrollment_type': ['student', 'teacher'],
         'enrollment_term_id': term_id,
-        'per_page': 100,
-        'page': 1
+        'per_page': 100
     }
 
-    slim_course_dicts = []
+    # Make first course request
+    page_num = 1
+    logger.info(f'Course Page Number: {page_num}')
+    response = make_request_using_api_utils(url_ending_with_scope, params)
+    all_course_data = json.loads(response.text)
+    slim_course_dicts = slim_down_course_data(all_course_data)
     more_pages = True
+
     while more_pages:
-        logger.info(f"Course Page Number: {params['page']}")
-        all_course_data = make_request_using_api_utils(f'{API_SCOPE_PREFIX}/{url_ending}', params)
-        if len(all_course_data) > 0:
+        next_params = API_UTIL.get_next_page(response)
+        if next_params:
+            page_num += 1
+            logger.info(f'Course Page Number: {page_num}')
+            response = make_request_using_api_utils(url_ending_with_scope, next_params)
+            all_course_data = json.loads(response.text)
             slim_course_dicts += slim_down_course_data(all_course_data)
```

Comment:
The coding within line 85-88 and line 95-98 are very similar. Could them be refactored into a single function?

Reply:
I could, yes. However, it seems it wouldn't be terribly useful outside of this function, since it's using this `slim_down_course_data` function.

Part of the reason for these changes though was to generalize the existing request function so that it only returned a response (thus making it necessary to do the `json.loads(response.text)` operation). Hopefully this means we can remove some of the duplicated functionality @pushyamig had created for her PR.

Reply:
sounds good to me! I will close this request then.